src/main/scala/outwatch/dom/VDomModifier.scala

```diff
@@ -51,25 +51,21 @@ final case object EmptyVDomModifier extends VDomModifier
 
 
 object VDomModifier {
-  final implicit class StringNode(string: String) extends VNode {
-    val asProxy = VNodeProxy.fromString(string)
+  final implicit class StringNode(val string: String) extends AnyVal with VNode {
+    def asProxy = VNodeProxy.fromString(string)
   }
 
   implicit def OptionIsEmptyModifier(opt: Option[VDomModifier]): VDomModifier = opt getOrElse EmptyVDomModifier
 
-  final case class VTree(nodeType: String,
-                   children: Seq[VNode],
-                   attributeObject: DataObject
-                  ) extends VNode {
+  final case class VTree(nodeType: String, modifiers: Vector[VDomModifier]) extends VNode {
+    import helpers.DomUtils
 
+    def asProxy = {
+      val (children, attributeObject) = DomUtils.extractChildrenAndDataObject(modifiers)
+      val childProxies: js.Array[VNodeProxy] = children.map(_.asProxy)(breakOut)
+      h(nodeType, attributeObject, childProxies)
+    }
 
-    lazy val childProxies: js.Array[VNodeProxy] = children.map(_.asProxy)(breakOut)
-
-    val asProxy = h(nodeType, attributeObject, childProxies)
-
+    def apply(args: VDomModifier*):VNode = VTree(nodeType, modifiers ++ args)
```

Comment:
Can we move this to `VNode` so we don't have to change all the type signatures? We could just pattern match, and it could be a NoOp on `StringNode`.

Reply:
`apply()` on `StringNode` does not make sense, so I think it is better to encode that in the type. Having that in `VNode` and a NoOp on `StringNode` could lead to confusion.